File: test/aidbox_sdk/snapshots/dotnethl7-fhir-r4-core/ConceptMap.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class ConceptMap : DomainResource, IResource
{
    public string? Description { get; set; }
    public string? SourceCanonical { get; set; }
    public string? Date { get; set; }
    public string? TargetUri { get; set; }
    public ConceptMap_Group[]? Group { get; set; }
    public string? Publisher { get; set; }
    public Base.CodeableConcept[]? Jurisdiction { get; set; }
    public string? Purpose { get; set; }
    public string? Name { get; set; }
    public Base.UsageContext[]? UseContext { get; set; }
    public string? Copyright { get; set; }
    public bool? Experimental { get; set; }
    public object? Source    
    {
        get
        {
            if (SourceCanonical is not null)
            {
                return SourceCanonical;
            }
    
            if (SourceUri is not null)
            {
                return SourceUri;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(string))
            {
                SourceCanonical = (string)value;
                return;
            }
    
            if (value?.GetType() == typeof(string))
            {
                SourceUri = (string)value;
                return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }
    public string? Title { get; set; }
    public string? TargetCanonical { get; set; }
    public required string Status { get; set; }
    public string? SourceUri { get; set; }
    public string? Url { get; set; }
    public Base.Identifier? Identifier { get; set; }
    public object? Target    
    {
        get
        {
            if (TargetUri is not null)
            {
                return TargetUri;
            }
    
            if (TargetCanonical is not null)
            {
                return TargetCanonical;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(string))
            {
                TargetUri = (string)value;
                return;
            }
    
            if (value?.GetType() == typeof(string))
            {
                TargetCanonical = (string)value;
                return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }
    public string? Version { get; set; }
    public Base.ContactDetail[]? Contact { get; set; }

    public class ConceptMap_Group_Element_Target_DependsOn : BackboneElement
    {
        public required string Value { get; set; }
        public string? System { get; set; }
        public string? Display { get; set; }
        public required string Property { get; set; }
    }

    public class ConceptMap_Group_Element_Target : BackboneElement
    {
        public string? Code { get; set; }
        public string? Comment { get; set; }
        public string? Display { get; set; }
        public Base.ResourceReference[]? Product { get; set; }
        public ConceptMap_Group_Element_Target_DependsOn[]? DependsOn { get; set; }
        public required string Equivalence { get; set; }
    }

    public class ConceptMap_Group_Element : BackboneElement
    {
        public string? Code { get; set; }
        public ConceptMap_Group_Element_Target[]? Target { get; set; }
        public string? Display { get; set; }
    }

    public class ConceptMap_Group_Unmapped : BackboneElement
    {
        public string? Url { get; set; }
        public string? Code { get; set; }
        public required string Mode { get; set; }
        public string? Display { get; set; }
    }

    public class ConceptMap_Group : BackboneElement
    {
        public string? Source { get; set; }
        public string? Target { get; set; }
        public required ConceptMap_Group_Element[] Element { get; set; }
        public ConceptMap_Group_Unmapped? Unmapped { get; set; }
        public string? SourceVersion { get; set; }
        public string? TargetVersion { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/ConceptMap.py
from typing import Optional, List
from base import UsageContext
from base import ContactDetail
from base import CodeableConcept
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

class ConceptMap_Group_Element_Target_DependsOn(BackboneElement):
    value: str
    system: Optional[str] = None
    display: Optional[str] = None
    property: str

class ConceptMap_Group_Element_Target(BackboneElement):
    code: Optional[str] = None
    comment: Optional[str] = None
    display: Optional[str] = None
    product: Optional[List[Reference]] = None
    depends_on: Optional[List[ConceptMap_Group_Element_Target_DependsOn]] = None
    equivalence: str

class ConceptMap_Group_Element(BackboneElement):
    code: Optional[str] = None
    target: Optional[List[ConceptMap_Group_Element_Target]] = None
    display: Optional[str] = None

class ConceptMap_Group_Unmapped(BackboneElement):
    url: Optional[str] = None
    code: Optional[str] = None
    mode: str
    display: Optional[str] = None

class ConceptMap_Group(BackboneElement):
    source: Optional[str] = None
    target: Optional[str] = None
    element: list[ConceptMap_Group_Element] = []
    unmapped: Optional[ConceptMap_Group_Unmapped] = None
    source_version: Optional[str] = None
    target_version: Optional[str] = None

class ConceptMap(DomainResource):
    description: Optional[str] = None
    source_canonical: Optional[str] = None
    date: Optional[str] = None
    target_uri: Optional[str] = None
    group: Optional[List[ConceptMap_Group]] = None
    publisher: Optional[str] = None
    jurisdiction: Optional[List[CodeableConcept]] = None
    purpose: Optional[str] = None
    name: Optional[str] = None
    use_context: Optional[List[UsageContext]] = None
    copyright: Optional[str] = None
    experimental: Optional[bool] = None
    title: Optional[str] = None
    target_canonical: Optional[str] = None
    status: str
    source_uri: Optional[str] = None
    url: Optional[str] = None
    identifier: Optional[Identifier] = None
    version: Optional[str] = None
    contact: Optional[List[ContactDetail]] = None